docs/proposals/25-namespaced-cloud-profiles.md

```diff
@@ -0,0 +1,338 @@
+spec:
+  parent:
+    kind: CloudProfile
+    name: aws-central-cloud-profile
+  kubernetes:
+    versions:
+      - version: 1.28.6
+        expirationDate: 2024-06-06T01:02:03Z
+  machineImages:
+    - name: suse-chost
+      versions:
+        - version: 16.4
+          expirationDate: 2023-08-8T23:59:59Z
+  machineTypes: 
+    - name: m5.xlarge
+      cpu: "8"
+      gpu: "0"
+      memory: 16Gi
+  volumeTypes:
+    - name: ab6
+      class: premium
+      usable: true
+status:
+  cloudProfile:
+    apiVersion: core.gardener.cloud/v1beta1
+    kind: CloudProfile
+    spec:
+      type: aws
+      kubernetes:
+        versions:
+          - version: 1.27.1
+          - version: 1.26.3
+          - version: 1.25.8
+          - version: 1.24.6
+          - version: 1.28.6
+            expirationDate: 2024-06-06T01:02:03Z
+      machineImages:
+        - name: suse-chost
+          versions:
+            - version: 16.4
+              expirationDate: 2023-08-8T23:59:59Z
+            - version: 15.4
+            - version: 14.4
+            - version: 13.6
+      machineTypes: 
+        - name: m5.large
+          cpu: "4"
+          gpu: "0"
+          memory: 8Gi
+        - name: m5.xlarge
+          cpu: "8"
+          gpu: "0"
+          memory: 16Gi
+      volumeTypes:
+        - name: ab6
+          class: premium
+          usable: true
+        - name: gp3
+          class: standard
+          usable: true
+```
+
+The rendering is done by a new custom controller registered to the `gardener-controller-manager`. Merge conflicts can not arise during the merge process as they are caught by static validation and an admission plugin for validating the `NamespacedCloudProfile` object.
+
+### Custom RBAC verb
+
+To prevent users from entering arbitrary values in the `kubernetes` and `machineImages` fields, two custom RBAC verbs may be introduced. It can then be checked if the user that is creating or updating the `kubernetes` or `machineImages` field is authorized to do so.
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: namespaced-cloud-profile-modify-special-fields
+rules:
+  - apiGroups: [""]
+    resources: ["NamespacedCloudProfile"]
+    verbs: ["modify-spec-kubernetes", "modify-spec-machineImages"]
+```
+
+### Adjusting `Shoot`s `cloudProfileName` field
+
+Since `CloudProfiles` and `NamespacedCloudProfiles` are separate API objects, a `CloudProfile` could have the same name as a `NamespacedCloudProfile`. When the Shoot then references the `CloudProfile` two possible profiles could match on the reference. To solve this issue, the `Shoot` object should be extended with a `cloudProfile` field that specifies both the `name` and the `kind` of the referenced `CloudProfile`. However, the existing `cloudProfileName` field should remain intact and default to a `CloudProfile` for a smooth migration. Sane default will also be defined to allow for full backward compatibility with the `cloudProfileName` field.
```

Comment:
```suggestion
Since `CloudProfiles` and `NamespacedCloudProfiles` are separate API objects, a `CloudProfile` could have the same name as a `NamespacedCloudProfile`. When the Shoot references the `CloudProfile`, two possible profiles could match on the reference. To solve this issue, the `Shoot` object should be extended with a `cloudProfile` field that specifies both the `name` and the `kind` of the referenced `CloudProfile`. However, the existing `cloudProfileName` field should remain intact and default to a `CloudProfile` for a smooth migration. Same default will also be defined to allow for full backward compatibility with the `cloudProfileName` field.
```